File: bayeosgatewayclient/bayeosgatewayclient/testtimedelay.py
"""Script to test the time delay during writing."""

from time import time, sleep, strptime, strftime, mktime
from bayeosgatewayclient import BayEOSWriter, BayEOSSender, bayeos_argparser
# Fetch input arguments

args = bayeos_argparser('Measures time delay between two frames.')

WRITER_SLEEP = float(args.writer_sleep)
MAX_CHUNK = args.max_chunk
NAME = args.name
PATH = args.path + '/' + NAME + '/'

print 'name to appear in Gateway is', NAME
print 'max-chunk is', MAX_CHUNK, 'byte'
print 'writer sleep time is', WRITER_SLEEP, 'sec'
print 'path to store writer files is', PATH

# URL = 'http://bayconf.bayceer.uni-bayreuth.de/gateway/frame/saveMatrix'
URL = 'http://bayconf.bayceer.uni-bayreuth.de/gateway/frame/saveFlat'

# init writer and sender
writer = BayEOSWriter(PATH, MAX_CHUNK)
writer.save_msg('Writer was started.')

sender = BayEOSSender(PATH, NAME + '-WS' + str(WRITER_SLEEP) + '-M' + str(MAX_CHUNK), URL, 'import', 'import')
sender.start(2)

# start measurement
today = mktime(strptime(strftime('%Y-%m-%d'), '%Y-%m-%d'))
start = time()
t_run = time() - start

while t_run <= 1000:
    t = time()
    t_run = t - start
    writer.save([t_run, t - today], value_type=0x21)
    sleep(WRITER_SLEEP)

File: tests/testtimedelay.php
<?php 
/*
 * Run the Sender in the background as thread
 * 
 * To run this script you have to install pthreads manually
 * 
 * 
 */
require_once 'BayEOSGatewayClient.php';
$path='/tmp/PHP-Delay-Test-ultrabook-tmp-WS0.05-M2560.0';
$name="PHP-Delay-Test-ultrabook-tmp-WS0.05-M2560.0";
$url="http://bayconf.bayceer.uni-bayreuth.de/gateway/frame/saveMatrix";
$max_chunk = 2560;
$writer_sleep = 0.05;

//Create a BayEOSWriter
$w = new BayEOSWriter($path, $max_chunk);
$w->saveMessage("testtimedelay.php started");

// //Create a BayEOSSender and start it as thread
// class SenderThread extends Thread {
// 	public function run() {
// 		$s = new BayEOSSender($GLOBALS['path'],$GLOBALS['name'],$GLOBALS['url']);
// 	}
// }
// $s_thread=new SenderThread();
// $s_thread->start();
//mktime(date('H'), date('i'), date('s'), date('m'), date('d'), date('y'))

$today = mktime(0, 0, 0, date('m'), date('d'), date('y'));
$start = microtime(true);
$t_run = microtime(true) - $start;
echo $today, PHP_EOL;
echo $start, PHP_EOL;
echo $t_run, PHP_EOL;
echo (microtime(true)-$today);

while($t_run <= 1000){
	$t = microtime(true);
	$t_run = $t - $start;
	$w->save(array($t_run, ($t-$today)));
	sleep($writer_sleep);
}

$s = new BayEOSSender($path,$name,$url);

while($t_run <= 1500){
	$c = $s->send();
	$t_run = $t - $start;
	if($c){
		echo "Successfully sent $c frames\n";
	}
	sleep(0.1);	
}


?>
